arm-servicebus/2015-08-01/swagger/servicebus.json

```diff
@@ -2642,6 +2886,213 @@
         "modelAsString": false
       },
       "description": "Entity status."
+    },
+    "RuleListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RuleResource"
+          },
+          "description": "Result of the List Rules operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results. Not empty if Value contains incomplete list of rules"
+        }
+      },
+      "description": "The response of the List rule operation."
+    },
+    "RuleCreateOrUpdateParameters": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Rule name."
+        },
+        "location": {
+          "type": "string",
+          "description": "Location of the resource."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RuleProperties"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "description": "Parameters supplied to the CreateOrUpdate operation."
+    },
+    "RuleResource": {
+      "properties": {
+        "filter": {
+          "$ref": "#/definitions/Filter",
+          "description": "Describes a filter expression that is evaluated against a BrokeredMessage."
+        },
+        "action": {
+          "$ref": "#/definitions/RuleAction",
+          "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+        },
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Creation time of the rule.",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Description of Rule Resource."
+    },	
+	"RuleResource_SQLFilter": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RuleProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Description of Rule Resource."
+    },
+    "Filter": {
+      "type": "object",
+      "properties": {        
+        "filterType": {
+		  "type": "string",
+		  "enum": [
+			"SqlFilter",
```

Comment:
So....., if I understand correctly, filter is a base class referenced in RuleProperties which is inherited by appropriate filter types and "filterType" is the identifier for the particular inherited type. In swagger world this boils down directly to the concept of ```discriminator``` Please refer http://swagger.io/specification/ and https://github.com/Azure/azure-rest-api-specs/blob/400a8125ba30578827f4df29cc6428617e293b78/arm-commerce/2015-06-01-preview/swagger/commerce.json#L360